### PR TITLE
[add] py-cylc-uiserver: new recipe

### PR DESCRIPTION
**Pull request by LydDeb**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the home page and version sha256.  Adding python maintainer.

File: var/spack/repos/builtin/packages/py-cylc-uiserver/package.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    homepage = "https://github.com/cylc/cylc-uiserver/"
+    pypi = "cylc-uiserver/cylc-uiserver-1.3.0.tar.gz"
+
+    maintainers("LydDeb")
+
+    version("1.3.0", sha256="f3526e470c7ac2b61bf69e9b8d17fc7a513392219d28baed9b1166dcc7033d7a")
+
+    depends_on("python@3.7:3.9", type=("build", "run"))
+    depends_on("py-wheel", type="build")
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-cylc-uiserver/package.py:19]: `py-wheel` is automatically added by the `PythonPackage` class and you only need to add it again as dependency if a specific version is needed.

File: var/spack/repos/builtin/packages/py-cylc-uiserver/package.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    """Cylc UI Server."""
+
+    homepage = "https://github.com/cylc/cylc-uiserver/"
+    pypi = "cylc-uiserver/cylc-uiserver-1.3.0.tar.gz"
+
+    maintainers("LydDeb")
+
+    version("1.3.0", sha256="f3526e470c7ac2b61bf69e9b8d17fc7a513392219d28baed9b1166dcc7033d7a")
+
+    depends_on("python@3.7:3.9", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-cylc-uiserver/package.py]: Where do get there upper restriction from?

File: var/spack/repos/builtin/packages/py-cylc-uiserver/package.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    depends_on("py-cylc-flow@8.2", type=("build", "run"))
+    depends_on("py-ansimarkup@1.0.0:", type=("build", "run"))
+    depends_on("py-graphene", type=("build", "run"))
+    depends_on("py-graphene-tornado@2.6", type=("build", "run"))
+    depends_on("py-graphql-ws@0.4.4", type=("build", "run"))
+    depends_on("py-jupyter-server@1.10.2:1", type=("build", "run"))
+    depends_on("py-requests", type=("build", "run"))
+    depends_on("py-tornado@6.1.0:", type=("build", "run"))
+    depends_on("py-traitlets@5.2.1:", type=("build", "run"))
+    depends_on("py-pyzmq", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-cylc-uiserver/package.py:30]: Missing dependency on `py-graphql-core

**Review by manuelakuhn — COMMENTED** [review]
Depends on #39986, #39989, #39988, #39985, #39991

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**